h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -270,8 +276,10 @@ public static ParseSetup guessSetup( Key[] fkeys, ParseSetup userSetup ) {
     if (ice instanceof Frame && ((Frame) ice).vec(0) instanceof UploadFileVec) {
       t._gblSetup._chunk_size = FileVec.DFLT_CHUNK_SIZE;
     } else {
-      t._gblSetup._chunk_size = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
-              Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int calculatedChunkSize = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
+          Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int finalChunkSize = Math.max(t._gblSetup.dataOffset + 2, calculatedChunkSize);
```

Comment:
There is a sample test case for that.
The first one will define the chunk size. When the next one is opened, it will be too late to change it anyway. (I'm planning to fix this later on).
Generally speaking the impact of dataOffset is not so categorical; on many occasions the parsers will work even when the first dataOffset is 0. 
Good point anyway; I'll have to concoct a test case where the header is huge in the second file. It will probably fail with the current code. There are more cases that will fail, actually.